src/core/public/rendering/rendering_service.tsx

```diff
@@ -41,53 +39,28 @@ interface StartDeps {
  * @internalRemarks Currently this only renders Chrome UI. Notifications and
  * Overlays UI should be moved here as well.
  *
- * @returns a DOM element for the legacy platform to render into.
- *
  * @internal
  */
 export class RenderingService {
-  start({
-    application,
-    chrome,
-    injectedMetadata,
-    overlays,
-    targetDomElement,
-  }: StartDeps): RenderingStart {
+  start({ application, chrome, overlays, targetDomElement }: StartDeps) {
     const chromeUi = chrome.getHeaderComponent();
     const appUi = application.getComponent();
     const bannerUi = overlays.banners.getComponent();
 
-    const legacyMode = injectedMetadata.getLegacyMode();
```

Comment:
could you remove `legacyMode`
https://github.com/elastic/kibana/blob/4c810be33595d478c6eb802170d0644b32e62898/src/core/server/rendering/rendering_service.tsx#L85

https://github.com/elastic/kibana/blob/8598db5afd0867421ec24d9fff2e77fa5412d85e/src/core/public/chrome/ui/header/header.test.tsx#L53